File: client/src/interfaces/game-state.response.ts
export type Role = "Troop" | "King";
export type Color = "White" | "Black";

export interface Piece {
    color: Color;
    role: Role;
}

export interface Point {
    x: number;
    y: number;
}

export interface Square {
    point: Point;
    piece: Piece | null;
}

export interface Board {
    rows: number;
    cols: number;
    squares: Square[];
}

export interface Status {
    currentPlayer: string;
    gameStatus: string;
    winner: string | null;
}

export interface GameStateResponse {
    board: Board;
    status: Status;
}

export interface MoveRequest {
    from: Point;
    to: Point;
}

export type HighlightedMove = {
    from: Point;
    valid: {
        to: Point;
        enemyCaptured: Point | null;
    }[];
};

File: server/Dtos/GameStateResponse.cs
using CheckerboardGameApp.Models;

namespace CheckerboardGameApp.Dtos;

public record GameStateResponse
{
    public required GameStatusInfo Status { get; set; }
    public required BoardInfo Board { get; set; }
}

public record GameStatusInfo
{
    public required string CurrentPlayer { get; set; }
    public required string GameStatus { get; set; }
    public required object WhitePlayer { get; set; }
    public required object BlackPlayer { get; set; }
    public object? Winner { get; set; }
}

public record BoardInfo
{
    public required int Rows { get; set; }
    public required int Cols { get; set; }
    public required List<Square> Squares { get; set; }
}
